buildspec.yml

```diff
@@ -2,61 +2,69 @@ version: 0.2
+          GITHUB_BRANCH="$(git branch -a --contains HEAD | sed -n 2p | awk '{ printf $1 }')";
+          export GITHUB_BRANCH=${GITHUB_BRANCH#remotes/origin/};
+        fi
       - COMMIT_HASH=$(echo $CODEBUILD_RESOLVED_SOURCE_VERSION | cut -c 1-7)
-      - GITHUB_BRANCH=${CODEBUILD_WEBHOOK_HEAD_REF#refs/heads/*}
-      - TODAY=$(date '+%Y%m%d')
-      - |-
-           if [ "${CODEBUILD_WEBHOOK_TRIGGER}" = pr/* ]; then
-             PULL_REQUEST=${CODEBUILD_WEBHOOK_TRIGGER#pr/*}
-           fi
-      - |-
-           if [ "${GIT_BRANCH}" != master  ]; then
-               LATEST=${GITHUB_BRANCH:-staging}
-           else
-               LATEST=latest
-           fi
-      - echo "=========== VERSIONS ============="
-      - python --version
-      - python2 --version
-      - bash --version
-      - echo "=========== VARIABLES ============="
-      - echo CODEBUILD_RESOLVED_SOURCE_VERSION=${CODEBUILD_RESOLVED_SOURCE_VERSION}
-      - echo CODEBUILD_SOURCE_VERSION=${CODEBUILD_SOURCE_VERSION}
-      - echo CODEBUILD_WEBHOOK_BASE_REF=${CODEBUILD_WEBHOOK_BASE_REF}
-      - echo CODEBUILD_WEBHOOK_HEAD_REF=${CODEBUILD_WEBHOOK_HEAD_REF=}
-      - echo CODEBUILD_WEBHOOK_TRIGGER=${CODEBUILD_WEBHOOK_TRIGGER}
-      - echo CODEBUILD_SOURCE_VERSION=${CODEBUILD_SOURCE_VERSION}
-      - echo CODEBUILD_BUILD_ID=${CODEBUILD_BUILD_ID}
-      - echo CODEBUILD_SOURCE_REPO_URL=${CODEBUILD_SOURCE_REPO_URL}
-      - echo COMMIT_HASH=${COMMIT_HASH}
-      - echo PULL_REQUEST=${PULL_REQUEST}
-      - echo GITHUB_BRANCH=${GITHUB_BRANCH}
-      - echo "==================================="
+      - echo "GITHUB_BRANCH=${GITHUB_BRANCH} COMMIT_HASH=${COMMIT_HASH}"
+      - make clean setup
+
   build:
     commands:
       - echo Build started on $(date)
-      - . ./rc_ci && make cleanall all
+      - export DOCKER_IMG_TAG_BASE=${IMAGE_BASE_NAME}:${GITHUB_BRANCH}
+      - export DOCKER_IMG_TAG_BRANCH=${DOCKER_IMG_TAG_BASE}.${COMMIT_HASH}
+      - export DOCKER_IMG_TAG_LATEST=${DOCKER_IMG_TAG_BASE}.latest
+      - echo "Building docker image with tags ${DOCKER_IMG_TAG_BRANCH} and ${DOCKER_IMG_TAG_LATEST}"
+      - docker build -t ${DOCKER_IMG_TAG_BRANCH} -t ${DOCKER_IMG_TAG_LATEST} .
 
   post_build:
     commands:
-      - make testunit
-      - make testintegration
+      - make lint
+      - mkdir -p ${TEST_REPORT_DIR}
+      - export TEST_REPORT_DIR=${TEST_REPORT_DIR}
+      - export TEST_REPORT_FILE=${TEST_REPORT_FILE}
+      - make test
+      # Only push image to dockerhub for merges on develop and master
+      # by not specifying one of the two tagging above but the common part the two tags will be
+      # pushed together (e.g. swisstopo/service-alti:develop.1234567 and swisstopo/service-alti:develop.latest
+      # if we push swisstopo/service-alti:develop the two tags will be pushed together)
+      - |
+        if [ "${GITHUB_BRANCH}" = "master" -o "${GITHUB_BRANCH}" = "develop" ]; then
+          docker push ${DOCKER_IMG_TAG_BASE}
```

Comment:
Should not it be here `${DOCKER_IMG_TAG_BRANCH}` or better to say `${DOCKER_IMG_TAG_HASH/COMMIT}` ?

Reply:
After testing with qrcode local build and looking at the documentation, the correct way here would be to call `docker push` twice as follow:
```
docker push ${DOCKER_IMG_TAG_HASH/COMMIT}
docker push ${DOCKER_IMG_TAG_LATEST}
```

Reply:
so two pushes, not possible to do it in one go. Thanks for looking into it